src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,181 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DataColumn;
+
+/**
+ * Defines tax rules grid
+ */
+final class TaxRuleGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    public const GRID_ID = 'tax_rules';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Tax Rules', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        return (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_tax_rule',
+                    ])
+            )
+            ->add(
+                (new DataColumn('country'))
+                    ->setName($this->trans('Country', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'country_name',
+                    ])
+            )
+            ->add(
+                (new DataColumn('state'))
+                    ->setName($this->trans('State', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'state_name',
+                    ])
+            )
+            ->add(
+                (new DataColumn('zipcode'))
+                    ->setName($this->trans('Zip/Postal code', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'zipcode',
+                    ])
+            )
+            ->add(
+                (new DataColumn('behavior'))
+                    ->setName($this->trans('Behavior', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'behavior',
+                    ])
+            )
+            ->add(
+                (new DataColumn('rate'))
+                    ->setName($this->trans('Tax', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'rate',
+                    ])
+            )
+            ->add(
+                (new DataColumn('description'))
+                    ->setName($this->trans('Description', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'description',
+                    ])
+            )
+            ->add(
+                (new ActionColumn('actions'))
+                    ->setName($this->trans('Actions', [], 'Admin.Global'))
+                    ->setOptions([
+                        'actions' => (new RowActionCollection())
+                        /*
```

Comment:
would be nice to have a comment (preferable a todo reminder) of why these are commented out (I suspect its just expected to be done in separate PR).
Not blocking though (unless you have other feedback to fix).

Reply:
It's my next PR.